platformio/src/client_utils.cpp

```diff
@@ -57,33 +58,57 @@
 wl_status_t startWiFi(int &wifiRSSI)
 {
   WiFi.mode(WIFI_STA);
-  Serial.printf("Connecting to '%s'", WIFI_SSID);
-  WiFi.begin(WIFI_SSID, WIFI_PASSWORD);
 
-  // timeout if WiFi does not connect in 10s from now
-  unsigned long timeout = millis() + 10000;
-  wl_status_t connection_status = WiFi.status();
+  WiFiManager wifi_manager;
+  wifi_manager.setConnectTimeout(10);
 
-  while ((connection_status != WL_CONNECTED) && (millis() < timeout))
+  // Turn off config AP fallback to save on battery on random errors
+  wifi_manager.setEnableConfigPortal(false);
+  if (wifi_manager.autoConnect(WIFI_AP_SSID)) 
   {
-    Serial.print(".");
-    delay(50);
-    connection_status = WiFi.status();
+    wifiRSSI = WiFi.RSSI(); // get WiFi signal strength now, because the WiFi
+                            // will be turned off to save power!
+    Serial.println("IP: " + WiFi.localIP().toString());
   }
-  Serial.println();
+  return WiFi.status();
+} // startWiFi
+
+/* Callback when WiFi config AP is started
+ * Takes in a WiFiManager instance. 
+ */
+void configModeCallback(WiFiManager *myWiFiManager)
+{
+  Serial.println("Entered Configuration Mode");
+  Serial.print("Config SSID: ");
+  Serial.println(myWiFiManager->getConfigPortalSSID());
+  Serial.print("Config IP Address: ");
+  Serial.println(WiFi.softAPIP());
+}
 
-  if (connection_status == WL_CONNECTED)
+/* Power-on and start WiFi config AP.
+ * Takes int parameter to store WiFi RSSI, or “Received Signal Strength
+ * Indicator"
+ *
+ * Returns WiFi status.
+ */
+wl_status_t configureWiFi(int &wifiRSSI)
```

Comment:
This is the code that leverages the libraries ability to spawn up a config AP that allows you to configure the network by connecting to the config AP.